### PR TITLE
Add URI paths for cluster stats filtering.

### PR DESCRIPTION
**Pull request by SwethaGuptha**
### Description
Documentation change for new cluster stats URI path for filtering requested metrics.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8527

### Version
_List the OpenSearch version to which this PR applies, e.g. 2.18

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by shwetathareja — APPROVED** [review]

**Comment by SwethaGuptha** [comment]
@Naarcha-AWS Can you please let me know what would be the next steps for this PR or any action is pending on me?

File: _api-reference/cluster-api/cluster-stats.md [hunk 1]
```diff
@@ -30,11 +32,54 @@ All parameters are optional.
 Parameter | Type | Description
 :--- | :--- | :---
 &lt;node-filters&gt; | List | A comma-separated list of [node filters]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/index/#node-filters) that OpenSearch uses to filter results.
+metric | String | A comma-separated list of metric groups that are included in the response. For example, `jvm,fs`. See the following list of all metrics. Defaults to all metrics.
```
Comment (Naarcha-AWS) [inline _api-reference/cluster-api/cluster-stats.md]:
```suggestion
metric | String | A comma-separated list of metric groups to include in the response, for example, `jvm.fs`. Default is all metric groups.
```

**Review by Naarcha-AWS — APPROVED** [review]
Made my edits on top. Will pass this to editorial review.

**Comment by Naarcha-AWS** [comment]
> @Naarcha-AWS Can you please let me know what would be the next steps for this PR or any action is pending on me?

@SwethaGuptha: No other action is required on your part at this time! There's a chance we might have a couple of follow up questions about the PR, but we'll ping you directly on a comment if that's the case.

File: _api-reference/cluster-api/cluster-stats.md [hunk 2]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
 &lt;node-filters&gt; | List | A comma-separated list of [node filters]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/index/#node-filters) that OpenSearch uses to filter results.
+metric | String | A comma-separated list of [metric groups](#metric-groups), for example, `jvm,fs`. Default is all metric groups.
+index_metric | String | A comma-separated list of [index metric groups](#index-metric-groups), for example, `docs,store`. Default is all index metrics.
 
 
-  Although the `master` node is now called `cluster_manager` for version 2.0, we retained the `master` field for backwards compatibility. If you have a node that has either a `master` role or a `cluster_manager` role, the `count` increases for both fields by 1. To see an example node count increase, see the Response sample.
-   {: .note }
+Although the `master` terminology has been deprecated for `cluster_manager` after OpenSearch 2.0, we retained the `master` field for backwards compatibility. If you have a node that has either a `master` role or a `cluster_manager` role, the `count` increases for both fields by 1. To see an example node count increase, see the [example response](#example-response).
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:39]:
```suggestion
Although the term `master` was deprecated in favor of `cluster_manager` subsequent to OpenSearch 2.0, the `master` field was retained for backward compatibility. If you have a node that has either a `master` role or a `cluster_manager` role, the `count` increases for both fields by 1. For an example node count increase, see the [example response](#example-response).
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 3]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+{: .note }
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:40]: Is it versions "subsequent to 2.0", or should it be "in OpenSearch 2.0 and later"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 4]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+
+### Metric groups
+
+The following table lists all available metric groups.
+
+Metric | Description
+:--- |:----
+`indices` | Statistics about indexes in the cluster.
+`os` | Statistics about the host OS including load, and memory.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:49]:
```suggestion
`os` | Statistics about the host OS, including load and memory.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 5]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+`process` | Statistics about processes, including open file descriptors and CPU usage.
+`jvm` | Statistics about the JVM, including heap usage and threads.
+`fs` | Statistics about file system usage.
+`plugins` | Statistics about OpenSearch plugins integrated within the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:53]:
```suggestion
`plugins` | Statistics about OpenSearch plugins integrated with the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 6]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+`network_types` | The Transport and HTTP networks connected to the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:54]:
```suggestion
`network_types` | The transport and HTTP networks connected to the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 7]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+`discovery_type` | The method the nodes use to find other nodes inside the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:55]:
```suggestion
`discovery_type` | The method used by the nodes to find other nodes in the cluster.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 8]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+`packaging_types` | Information about each nodes’ OpenSearch distribution.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:56]:
```suggestion
`packaging_types` | Information about each node's OpenSearch distribution.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 9]
```diff
@@ -30,10 +32,57 @@ All parameters are optional.
...
+`ingest` | Statistics about ingest pipelines.
+
+### Index metric groups
+
+To filter the information returned for the `indices` metric, you can use specific `index_metric` values. You can use these only when you use the following query types:
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:61]:
```suggestion
To filter the information returned for the `indices` metric, you can use specific `index_metric` values. These values are only supported when using the following query types:
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 10]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
-nodes.process | The processes the returned nodes use.
-nodes.jvm | Statistics about the Java Virtual Machines in use.
-nodes.fs | The nodes' file storage.
-nodes.plugins | The OpenSearch plugins integrated within the nodes.
-nodes.network_types | The transport and HTTP networks within the nodes.
-nodes.discovery_type | The method the nodes use to find other nodes within the cluster.
-nodes.packaging_types | Information about the nodes' OpenSearch distribution.
-nodes.ingest | Information about the nodes' ingest pipelines/nodes, if there are any.
-total_time_spent | The total amount of download and upload time spent across all shards in the cluster when downloading or uploading from the remote store.
+`nodes` | How many nodes returned in the response.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:543]:
```suggestion
`nodes` | The number of nodes returned in the response.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 11]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`cluster_name` | The cluster's name.
+`cluster_uuid` | The cluster's uuid.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:545]:
```suggestion
`cluster_uuid` | The cluster's UUID.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 12]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`timestamp` | The Unix epoch time of when the cluster was last refreshed.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:546]:
```suggestion
`timestamp` | The Unix epoch time indicating when the cluster was last refreshed.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 13]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`status` | The cluster's health status.
+`indices` | Statistics about the indexes in the cluster.
+`indices.count` | How many indexes are in the cluster.
+`indices.shards` | Information about the cluster's shards.
+`indices.docs` | How many documents are still in the cluster and how many documents are deleted.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:551]:
```suggestion
`indices.docs` | The number of documents remaining in the cluster and the number of documents that were deleted.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 14]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`indices.store` | Information about the cluster's storage.
+`indices.fielddata` | Information about the cluster's field data
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:553]:
```suggestion
`indices.fielddata` | Information about the cluster's field data.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 15]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`indices.query_cache` | Data about the cluster's query cache.
+`indices.completion` | How many bytes in memory are used to complete operations.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:555]: "How many" => "The number of". "are" => "that were"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 16]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`indices.segments` | Information about the cluster's segments, which are small Lucene indexes.
+`indices.mappings` | Mappings within the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:557]:
```suggestion
`indices.mappings` | Mappings in the cluster.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 17]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`indices.analysis` | Information about analyzers used in the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:558]: Above: What about them? "Information about mappings in the cluster"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 18]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes` | Statistics about the nodes in the cluster.
+`nodes.count` | How many nodes were returned from the request.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:560]:
```suggestion
`nodes.count` | The number of nodes returned by the request.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 19]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.versions` | OpenSearch's version number.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:561]: "The OpenSearch version number"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 20]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.os` | Information about the operating systems used in the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:562]:
```suggestion
`nodes.os` | Information about the operating systems used by the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 21]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.process` | The processes the returned nodes use.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:563]:
```suggestion
`nodes.process` | The processes used by the returned nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 22]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.jvm` | Statistics about the Java Virtual Machines in use.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:564]:
```suggestion
`nodes.jvm` | Statistics about the JVMs in use.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 23]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.fs` | The nodes' file storage.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:565]: What about it? "Information about the node's file storage"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 24]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.plugins` | The OpenSearch plugins integrated within the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:566]:
```suggestion
`nodes.plugins` | The OpenSearch plugins integrated with the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 25]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.network_types` | The transport and HTTP networks within the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:567]:
```suggestion
`nodes.network_types` | The transport and HTTP networks connected to the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 26]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.discovery_type` | The method the nodes use to find other nodes within the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:568]:
```suggestion
`nodes.discovery_type` | The method used by the nodes to find other nodes in the cluster.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 6 above)
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:54]: "Statistics about the"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 27]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
-nodes.packaging_types | Information about the nodes' OpenSearch distribution.
-nodes.ingest | Information about the nodes' ingest pipelines/nodes, if there are any.
-total_time_spent | The total amount of download and upload time spent across all shards in the cluster when downloading or uploading from the remote store.
+`nodes` | How many nodes returned in the response.
+`cluster_name` | The cluster's name.
+`cluster_uuid` | The cluster's uuid.
+`timestamp` | The Unix epoch time of when the cluster was last refreshed.
+`status` | The cluster's health status.
+`indices` | Statistics about the indexes in the cluster.
+`indices.count` | How many indexes are in the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:549]:
```suggestion
`indices.count` | The number of indexes in the cluster.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 21 above)
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:563]: "A list of the"? "Information about the"?

File: _api-reference/cluster-api/cluster-stats.md (hunk 24 above)
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:566]: "A list of the"? "Information about the"?

File: _api-reference/cluster-api/cluster-stats.md (hunk 25 above)
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:567]: "A list of the"? "Information about the"?

File: _api-reference/cluster-api/cluster-stats.md [hunk 28]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.count` | How many nodes were returned from the request.
+`nodes.versions` | OpenSearch's version number.
+`nodes.os` | Information about the operating systems used in the nodes.
+`nodes.process` | The processes the returned nodes use.
+`nodes.jvm` | Statistics about the Java Virtual Machines in use.
+`nodes.fs` | The nodes' file storage.
+`nodes.plugins` | The OpenSearch plugins integrated within the nodes.
+`nodes.network_types` | The transport and HTTP networks within the nodes.
+`nodes.discovery_type` | The method the nodes use to find other nodes within the cluster.
+`nodes.packaging_types` | Information about the nodes' OpenSearch distribution.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:569]:
```suggestion
`nodes.packaging_types` | Information about each node's OpenSearch distribution.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 29]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.ingest` | Information about the nodes' ingest pipelines/nodes, if there are any.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:570]:
```suggestion
`nodes.ingest` | Information about the node's ingest pipelines/nodes, if there are any.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 23 above)
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:565]:
```suggestion
`nodes.fs` | The node's file storage.
```

**Review by natebower — CHANGES_REQUESTED** [review]
@Naarcha-AWS @SwethaGuptha Please see my comments and changes and tag me for approval when addressed. Thanks!

File: _api-reference/cluster-api/cluster-stats.md (hunk 6 above)
Comment (Naarcha-AWS) [reply _api-reference/cluster-api/cluster-stats.md:54]: This is would be a list of networks, not statistics.

File: _api-reference/cluster-api/cluster-stats.md (hunk 15 above)
Comment (Naarcha-AWS) [inline _api-reference/cluster-api/cluster-stats.md:555]:
```suggestion
`indices.completion` | The number of bytes in memory that were used to complete operations.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 19 above)
Comment (Naarcha-AWS) [inline _api-reference/cluster-api/cluster-stats.md:561]:
```suggestion
`nodes.versions` | The OpenSearch version number for each node.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 6 above)
Comment (natebower) [reply _api-reference/cluster-api/cluster-stats.md:54]: OK, then let's say "A list of the transport and HTTP networks connected to the nodes."

File: _api-reference/cluster-api/cluster-stats.md (hunk 23 above)
Comment (Naarcha-AWS) [inline _api-reference/cluster-api/cluster-stats.md:565]:
```suggestion
`nodes.fs` | Information about the nodes' file storage.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 24 above)
Comment (Naarcha-AWS) [reply _api-reference/cluster-api/cluster-stats.md:566]: A list of the.

File: _api-reference/cluster-api/cluster-stats.md (hunk 25 above)
Comment (Naarcha-AWS) [reply _api-reference/cluster-api/cluster-stats.md:567]:
```suggestion
`nodes.network_types` | A list of the transport and HTTP networks within the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md (hunk 3 above)
Comment (Naarcha-AWS) [reply _api-reference/cluster-api/cluster-stats.md:40]: Subsequent.

**Comment by Naarcha-AWS** [comment]
@natebower: Implemented your review. Let me know if I missed anything.

File: _api-reference/cluster-api/cluster-stats.md [hunk 30]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`indices.query_cache` | Data about the cluster's query cache.
+`indices.completion` | The number of bytes in memory that were used to complete operations.
+`indices.segments` | Information about the cluster's segments, which are small Lucene indexes.
+`indices.mappings` | Information about mappings in the cluster.
+`indices.analysis` | Information about analyzers used in the cluster.
+`nodes` | Statistics about the nodes in the cluster.
+`nodes.count` | The number of nodes returned by the request.
+`nodes.versions` | The OpenSearch version number for each node.
+`nodes.os` | Information about the operating systems used by the nodes.
+`nodes.process` | A list of process used by each node.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:563]:
```suggestion
`nodes.process` | A list of processes used by each node.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 31]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.jvm` | Statistics about the JVMs in use.
+`nodes.fs` | Information about the node's file storage.
+`nodes.plugins` | A list of the OpenSearch plugins integrated with the nodes.
+`nodes.network_types` | A list of the transport and HTTP networks within the nodes.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:567]:
```suggestion
`nodes.network_types` | A list of the transport and HTTP networks connected to the nodes.
```

File: _api-reference/cluster-api/cluster-stats.md [hunk 32]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes.discovery_type` | The method used by the nodes to find other nodes in the cluster.
```
Comment (natebower) [inline _api-reference/cluster-api/cluster-stats.md:568]: Above: Changed to match line 54.

**Review by natebower — APPROVED** [review]
@Naarcha-AWS LGTM with two noted changes. Thanks!

File: _api-reference/cluster-api/cluster-stats.md [hunk 33]
```diff
@@ -491,32 +540,32 @@ GET _cluster/stats/nodes/_cluster_manager
...
+`nodes` | Statistics about the nodes in the cluster.
+`nodes.count` | The number of nodes returned by the request.
+`nodes.versions` | The OpenSearch version number for each node.
+`nodes.os` | Information about the operating systems used by the nodes.
+`nodes.process` | A list of processes used by each node.
+`nodes.jvm` | Statistics about the JVMs in use.
+`nodes.fs` | Information about the node's file storage.
+`nodes.plugins` | A list of the OpenSearch plugins integrated with the nodes.
+`nodes.network_types` | A list of the transport and HTTP networks connected to the nodes.
+`nodes.discovery_type` | The method used by the nodes to find other nodes in the cluster.
```
Comment (Naarcha-AWS) [inline _api-reference/cluster-api/cluster-stats.md:568]:
```suggestion
`nodes.discovery_type` | A list of methods used by the nodes to find other nodes in the cluster.
```

**Pull request merged**